### PR TITLE
Mingw

### PR DESCRIPTION
**Pull request by xantares**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -451,25 +451,26 @@ IF(BUILD_USING_OTHER_LAPACK)
...
-        SET(MATH_LIBS_TO_USE_VN ${LAPACK_BEING_USED} pthreadVC2_x64)
+        SET(CMAKE_THREAD_LIBS_INIT pthreadVC2_x64)
     else()
-        SET(MATH_LIBS_TO_USE    ${LAPACK_BEING_USED} pthreadVC2)
-        SET(MATH_LIBS_TO_USE_VN ${LAPACK_BEING_USED} pthreadVC2)
+        SET(CMAKE_THREAD_LIBS_INIT pthreadVC2)
     endif()
 ENDIF()
 
+SET(MATH_LIBS_TO_USE    ${LAPACK_BEING_USED} ${CMAKE_THREAD_LIBS_INIT} ${REALTIME_LIB} ${DL_LIBRARY} ${MATH_LIBRARY})
```
Comment (chrisdembia) [inline CMakeLists.txt]: Can you make this line under 80 characaters?

File: CMakeLists.txt [hunk 2]
```diff
@@ -451,25 +451,26 @@ IF(BUILD_USING_OTHER_LAPACK)
...
-    SET(MATH_LIBS_TO_USE    ${LAPACK_BEING_USED} pthread ${REALTIME_LIB} dl m)
-    SET(MATH_LIBS_TO_USE_VN ${LAPACK_BEING_USED} pthread ${REALTIME_LIB} dl m)
-ELSE()
+    SET(MATH_LIBRARY m)
+    SET(DL_LIBRARY dl)
+    SET(CMAKE_THREAD_LIBS_INIT pthread)
+ENDIF()
+
+IF(MSVC)
     ## Assume Microsoft Visual Studio
```
Comment (chrisdembia) [inline CMakeLists.txt]: You can remove this comment now.

File: CMakeLists.txt [hunk 3]
```diff
@@ -451,25 +451,26 @@ IF(BUILD_USING_OTHER_LAPACK)
 "Basename of the actual Lapack library we're depending on; can't change here; see variable BUILD_USING_OTHER_LAPACK." FORCE)
 ENDIF(BUILD_USING_OTHER_LAPACK)
 
-IF(${CMAKE_C_COMPILER} MATCHES "cc" OR ${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
-    IF(APPLE)
-        SET(REALTIME_LIB)
-    ELSE()
+IF(UNIX)
```
Comment (chrisdembia) [inline CMakeLists.txt]: +1

File: CMakeLists.txt [hunk 4]
```diff
@@ -451,25 +451,26 @@ IF(BUILD_USING_OTHER_LAPACK)
...
+        SET(CMAKE_THREAD_LIBS_INIT pthreadVC2_x64)
     else()
-        SET(MATH_LIBS_TO_USE    ${LAPACK_BEING_USED} pthreadVC2)
-        SET(MATH_LIBS_TO_USE_VN ${LAPACK_BEING_USED} pthreadVC2)
+        SET(CMAKE_THREAD_LIBS_INIT pthreadVC2)
     endif()
 ENDIF()
 
+SET(MATH_LIBS_TO_USE    ${LAPACK_BEING_USED} ${CMAKE_THREAD_LIBS_INIT} ${REALTIME_LIB} ${DL_LIBRARY} ${MATH_LIBRARY})
+SET(MATH_LIBS_TO_USE_VN ${MATH_LIBS_TO_USE})
```
Comment (chrisdembia) [inline CMakeLists.txt]: +1

File: CMakeLists.txt [hunk 5]
```diff
@@ -451,25 +451,26 @@ IF(BUILD_USING_OTHER_LAPACK)
...
+IF(UNIX)
+    IF(NOT APPLE)
         SET(REALTIME_LIB rt)
     ENDIF()
-    SET(MATH_LIBS_TO_USE    ${LAPACK_BEING_USED} pthread ${REALTIME_LIB} dl m)
-    SET(MATH_LIBS_TO_USE_VN ${LAPACK_BEING_USED} pthread ${REALTIME_LIB} dl m)
-ELSE()
+    SET(MATH_LIBRARY m)
+    SET(DL_LIBRARY dl)
+    SET(CMAKE_THREAD_LIBS_INIT pthread)
```
Comment (chrisdembia) [inline CMakeLists.txt]: Nitpicky, but why name this differently? It's not a cmake variable.

**Comment by chrisdembia** [comment]
Can you add a comment in this section of the README [here](https://github.com/simbody/simbody/blob/master/README.md#installing), similar to the OSX comment, saying that on Windows you can also compile with mingw? Or should we leave this as an "undocumented" feature?

File: SimTKcommon/Geometry/src/DecorativeGeometry.cpp [hunk 6]
```diff
@@ -34,17 +34,17 @@ namespace SimTK {
...
+const Vec3 Gray    = Vec3(.5,.5,.5);
+const Vec3 Red     = Vec3( 1, 0, 0);
+const Vec3 Green   = Vec3( 0, 1, 0);
+const Vec3 Blue    = Vec3( 0, 0, 1);
+const Vec3 Yellow  = Vec3( 1, 1, 0);
+const Vec3 Orange  = Vec3( 1,.5, 0);
+const Vec3 Magenta = Vec3( 1, 0, 1);
+const Vec3 Purple  = Vec3(.5, 0,.5);
+const Vec3 Cyan    = Vec3( 0, 1, 1);
+const Vec3 White   = Vec3( 1, 1, 1);
```
Comment (sherm1) [inline SimTKcommon/Geometry/src/DecorativeGeometry.cpp]: How do these get exported from the DLL on Windows without `__declspec(dllexport)`? Great if they do but at one time they were undefined symbols without that macro there.

Comment (chrisdembia) [inline SimTKcommon/Geometry/src/DecorativeGeometry.cpp]: The appveyor test passes, and I believe at least some of those symbols are used in test cases and examples. So perhaps they are not needed?

Comment (sherm1) [inline SimTKcommon/Geometry/src/DecorativeGeometry.cpp]: Yeah -- could have been a bug in an older version of VC++. I see that the header files still have the EXPORT macro, and that should be all that is needed.

@xantares, were the duplicate EXPORT macros causing a failure when compiling with mingw?

**Comment by sherm1** [comment]
Please let me know when this is ready for final review and merge.

File: CMakeLists.txt (hunk 2 above)
Comment (xantares) [inline CMakeLists.txt]: ok

File: CMakeLists.txt (hunk 1 above)
Comment (xantares) [inline CMakeLists.txt]: yep

File: CMakeLists.txt (hunk 5 above)
Comment (xantares) [inline CMakeLists.txt]: usually it comes from findthreads.cmake

File: SimTKcommon/Geometry/src/DecorativeGeometry.cpp (hunk 6 above)
Comment (xantares) [inline SimTKcommon/Geometry/src/DecorativeGeometry.cpp]: yes, the export should only be in the header, never the cpp

**Comment by xantares** [comment]
I dont understand why the  AppVeyor build fails.I did not touch LAPACK_BEING_USED since a942056.

**Pull request closed without merging**